### PR TITLE
Retain and release parent device when creating and releasing sub-devices

### PR DESCRIPTION
**Pull request by Kerilk**
Unless I am mistaken, this should fix https://github.com/pocl/pocl/issues/1356

### CODE REVIEW

File: lib/CL/clCreateSubDevices.c [hunk 1]
```diff
@@ -185,6 +185,9 @@ POname(clCreateSubDevices)(cl_device_id in_device,
          new_devs[i]->core_start = sum;
        sum += new_devs[i]->core_count;
      }
+     for (i = 0; i < count_devices; ++i) {
```
Comment (pjaaskel) [inline lib/CL/clCreateSubDevices.c:188]: Can you run `tools/scripts/format-diff.sh` or `format-branch.sh` to fix the style (GNU C style in C files)?

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Outside the style issue, LGTM. I'll launch the CI.

File: lib/CL/clCreateSubDevices.c (hunk 1 above)
Comment (Kerilk) [reply lib/CL/clCreateSubDevices.c:188]: Will do, thanks for the review.

**Comment by Kerilk** [comment]
Formatting should be addressed.

**Pull request merged**